### ISSUE TITLE
Couldn't build proto file into descriptor pool: duplicate file name when using upb python implementation

### ISSUE THREAD

**Issue by Atheuz** [protocolbuffers/upb#1108]

**Comment by Atheuz** [comment]
@mkruskal-google yes, we're inserting our own descriptor into the generated code. My understanding is that this is something that's necessary for our confluent kafka schema registry lookups to work correctly.

But the base issue is that this works fine in the python implementation and the cpp implementation on 3.20.3, and it works fine on the python implementation on 4.21.12 as well. It's only the upb implementation that it fails on.

**Comment by mkruskal-google** [comment]
Note: this also seems to build just fine if you insert your custom descriptor into *both* generated files.  I can't explain why upb is *different* than the other implementations, but using two different descriptor files in two different protos that reference each other seems like it's asking for trouble

**Comment by Atheuz** [comment]
Yes, I'm aware.

The problem is that different implementations other than upb do not have an issue with this. I'm also trying to see if we can get rid of this internally, but for years this has worked fine for both cpp and python implementations and only with the 4.21 release has it broken with the switch over to upb as the default.

**Comment by Atheuz** [comment]
@mkruskal-google do you know of a way to force protoc/buf to not use the google/protobuf/descriptor_pb2.py but instead use our custom generated one without doing this hacky replacement? Maybe that'll fix our issue.

**Comment by Atheuz** [comment]
Here's the internal user story:

"I want to ensure our users can pick up their code bindings and be able to use them without any modifications
So that our developer experience is as seamless as possible, and communication with Kafka and the schema registry works correctly.

Notes:
We currently need to do some adjustment to our python code bindings before these are pushed to local PyPi. This will ensure producers/consumers will use matching schemas with the ones produced in the schema registry.

The main issue is that the protobuf library for python includes the google common code bindings as well, taking precedence over our own descriptor.proto. Unfortunately, the internal descriptor field seem to be different, and schema registry doesn't like that.

By doing some tweaks to our produced code bindings, however, we can make this work. Steps to modify the code bindings:

AFTER generating the python code bindings, we should:
- rename the google/protobuf folder into google_test/protobuf 
- any generated code binding importing from our google.protobuf needs to have their import changed, by pointing to google_test/protobuf - only the import of a file we own needs to be changed"

**Comment by mkruskal-google** [comment]
I'm not aware of any feature that supports swapping out descriptor.proto (other than a post-generation find/replace).  What are you trying to change in it?

**Comment by FullDataAlchemist** [comment]
> I'm not aware of any feature that supports swapping out descriptor.proto (other than a post-generation find/replace). What are you trying to change in it?

Hi, I also have the same issue here. I have a product that needs to be updated with proto schema, so I have a job for updating and replacing the schema. in this job I get the following error that 
```
TypeError: Couldn't build proto file into descriptor pool: duplicate symbol
```
